src/common/git.ts

```diff
@@ -0,0 +1,62 @@
+import { execFile } from 'child_process';
+import { promisify } from 'util';
+import { Repository } from '../typings/git';
+import Logger from './logger';
+
+const exec = promisify(execFile);
+
+/**
+ * Push a local branch to a remote.
+ *
+ * @param remote name of remote to push to
+ * @param {string?} localBranch name of local branch to push (default: current branch)
+ * @param {string?} remoteBranch name of remote branch (default: name of local branch)
+ */
+export async function push(
```

Comment:
we should be able to use the git extension API for this, it was added recently. would require bumping engine to 1.29

Reply:
yes we should do this.

Reply:
we can remove this file, not used anywhere